app/controllers/two_factor_authentication/webauthn_verification_controller.rb

```diff
@@ -33,6 +34,17 @@ def confirm
 
     private
 
+    def check_if_device_supports_platform_auth
+      return unless user_session.has_key?(:platform_authenticator_available)
+      if platform_authenticator? && !device_supports_webauthn_platform?
+        redirect_to login_two_factor_options_url
+      end
+    end
+
+    def device_supports_webauthn_platform?
+      user_session.delete(:platform_authenticator_available) == true
```

Comment:
It might be good to have some regression coverage here to enforce the intended effect of `delete` being to prevent the user from getting stuck in a loop if they explicitly choose F/T after being bumped back to the listing screen.